kernel/aster-nix/src/fs/inode_handle/dyn_cap.rs

```diff
@@ -126,6 +224,8 @@ impl FileLike for InodeHandle<Rights> {
 
     fn clean_for_close(&self) -> Result<()> {
         // Close does not guarantee that the data has been successfully saved to disk.
+        self.release_range_locks();
+        self.unlock_flock();
```

Comment:
I think the `unlock_flock()` is called when dropping the `InodeHandle`. 
The `clean_for_close` is called when we close a file, but the `InodeHandle` may not be dropped. This is a **big** difference.

Reply:
That's more reasonable. 

For `fcntl`:
> If a process closes any file descriptor referring to a file, then all of the process's locks on that file are released, regardless of the file descriptor(s) on which the locks were obtained.

For `flock`:
> lock is released either by an explicit LOCK_UN operation on any of these duplicate file descriptors, or when all such file descriptors have been closed.